lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -1590,17 +1590,53 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
         return success();
       });
 
-  patterns.onOp("Shape", 9,
-                [](OpBinder binder, ConversionPatternRewriter &rewriter) {
-                  Torch::ValueTensorType resultType;
-                  Value operand;
-                  if (binder.tensorOperand(operand) ||
-                      binder.tensorResultType(resultType))
-                    return failure();
-                  rewriter.replaceOpWithNewOp<Torch::Aten_ShapeAsTensorOp>(
-                      binder.op, resultType, operand);
-                  return success();
-                });
+  patterns.onOp(
+      "Shape", 9, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value operand;
+        if (binder.tensorOperand(operand) ||
+            binder.tensorResultType(resultType))
+          return failure();
+
+        auto inputType = dyn_cast<Torch::ValueTensorType>(operand.getType());
+        int64_t inputRank = inputType.getSizes().size();
+
+        auto shapeType = Torch::ValueTensorType::get(
+            binder.op->getContext(), SmallVector<int64_t>{inputRank},
+            resultType.getOptionalDtype());
+
+        Value shape = rewriter.create<Torch::Aten_ShapeAsTensorOp>(
+            binder.getLoc(), shapeType, operand);
+
+        int64_t start = 0;
+        int64_t end = -1;
+        if (binder.optionalS64IntegerAttr(start, "start", 0) &&
+            binder.optionalS64IntegerAttr(end, "end", -1)) {
```

Comment:
I suppose it's one or the other. the `optionalS64IntegerAttr` doesn't take a default and returns a failure if the op is missing the optional attr. `s64IntegerAttr` does take a default, but might fail for other reasons. 

Now that I think about it more, it might make sense to add `s64IntegerAttr` with the default values at the beginning (seperated by `||`), and remove this case because it would be covered below by the 
```cpp
if (start == 0 && end == -1) {
```

Reply:
My concern is that if these are separated by `&&`, then if it succeeds in binding the start, wouldn't it ignore the second check/get?